src/db/collections/ledger_update.rs

```diff
@@ -262,6 +262,58 @@ impl MongoDb {
             .find(filter, options)
             .await
     }
+
+    /// Calculates the current balances associated with the given [`Address`](crate::types::stardust::block::Address).
+    pub async fn calc_balances_for_address(&self, address: Address) -> Result<(u64, u64), Error> {
+        #[derive(Deserialize, Default)]
+        struct Amount {
+            amount: f64,
+        }
+
+        #[derive(Deserialize, Default)]
+        struct Balances {
+            total_balance: Amount,
+            locked_balance: Amount,
+        }
+
+        let balances = self
+            .0
+            .collection::<Balances>(LedgerUpdateDocument::COLLECTION)
+            .aggregate(
+                vec![
+                    doc! { "$match": {
+                        "address": &address, "is_spent": false,
```

Comment:
I'm afraid that this is not going to be enough to get the unspent outputs. This collection contains potentially two rows per output id:

* one for the creation event and
* one for the spending event

That means initially every output is inserted with `is_spent == false`. So to check if an output is really unspent we would have to:

1. Check if `is_spent == false`
2. Check if there exists not other row with the same output id where `is_spent == true`.

At this point, I'm thinking if it would make more sense to use the newly created `OutputDocument` collection for the balance computation.

Reply:
@Alexandcoats 👀

Reply:
You're right. I think I assumed the row would be updated